File: framework/cei_python3/impl/stringwrapper.py
class StringWrapper:
    def __init__(self):
        self.__items = list()
        self.__result = str()

    def add_string_array(self, string_array, trim):
        if string_array is None:
            return
        if trim:
            for item in list:
                self.__items.append(item.strip())
        else:
            self.__items.append(string_array)

    def append_string_item(self, value):
        self.__items.append(value)

    def combine_string_items(self, prefix, suffix, separator):
        self.__result = str()
        for value in self.__items:
            if len(self.__result) != 0:
                self.__result += separator
            self.__result += prefix + value + suffix

    def to_string(self):
        return self.__result

File: framework/cei_java/src/main/java/cei/impl/StringWrapper.java
package cei.impl;

import java.util.LinkedList;
import java.util.List;

public class StringWrapper {
    private final List<String> items = new LinkedList<>();
    private final StringBuilder result = new StringBuilder();

    public void addStringArray(List<String> stringArray, boolean trim) {
        if (stringArray == null) {
            return;
        }
        if (trim) {
            stringArray.forEach(item -> {items.add(item.trim());});
        } else {
            items.addAll(stringArray);
        }
    }

    public void appendStringItem(String item) {
        items.add(item);
    }

    public void combineStringItems(String prefix, String suffix, String separator) {
        result.setLength(0);
        items.forEach(item -> {
            if (result.length() != 0) {
                result.append(separator);
            }
            result.append(prefix + item + suffix);
        });
    }

    public String toNormalString() {
        return result.toString();
    }
}
